packages/hagrid/hagrid/deps.py

```diff
@@ -397,6 +397,17 @@ def is_apple_silicon() -> bool:
 ENVIRONMENT["apple_silicon"] = is_apple_silicon()
 
 
+def is_gitpod() -> bool:
+    return bool(os.environ.get("GITPOD_WORKSPACE_URL", None))
+
+
+def gitpod_url(port: Optional[int] = None) -> str:
+    workspace_url = os.environ.get("GITPOD_WORKSPACE_URL", "")
+    if port:
+        workspace_url.replace("https://", f"https://{port}-")
+    return workspace_url
```

Comment:
should this be
```suggestion
    return workspace_url.replace("https://", f"https://{port}-") if port else workspace_url
```
since `str.replace` is not inplace?

Reply:
yup great catch